plugins/net-forward.yaml

```
@@ -0,0 +1,34 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: net-forward
+spec:
+  version: "v1.0.3"       
+  platforms:
+  - selector:
+      matchLabels:
+        os: linux
+    uri: https://github.com/antitree/krew-net-forward/releases/download/v1.0.3/net-forward_v1.0.3.tar.gz
+    sha256: "645adb05d9ad68ecc91d9034cb80b1e2fe4762864323c12de0f2bc5afe20baa0"
+    bin: "net-forward"  
+  shortDescription: "Setup a true forwarder from your local machine to any TCP service accessible from within the cluster."
```

```suggestion
  shortDescription: "Forward to arbitrary TCP services within the cluster"
```
..or even
> Forward to arbitrary TCP services

..or maybe

> Like `kubectl port-forward` but to arbitrary services

---

IMO neither of these really explain much.

---

But the shortDescription doesn't really have to explain. The user just needs to get an idea of what she could achieve with this tool.

The `description` on the other hand should explain and the one here does it pretty well.

---

to me the problem with those two are:

> Forward to arbitrary TCP services

isn't that what port-forward does? forward tcp traffic to services on a local port?

> Like `kubectl port-forward` but to arbitrary services

ditto, "isn't this what port-forward is for"?

---

I see. Would that sound better?

> Like `kubectl port-forward` without restrictions

Otherwise I give up :man_shrugging:

---

From what I'm reading in the PR description this tool does this:

`Make local endpoints available in the cluster`

---

@antitree I think you have to clarify this. I hope you can take some inspiration from this discussion :laughing:

---

I'll update the description for sure but with `kubectl port-forward` there's a restriction that it must be a pod or a service in the cluster. For me, during security testings it's interesting to test scenarios of deploying a Pod in Namespace A and see if it can access a pod, service deployed in NamespaceB. Or the scenario where an external service (e.g. image registry) is using an IP allow list such that from a Pod, I can access the endpoint from the Pod. port-forward doesn't currently allow that.

---

Marking resolved as I attempted to make this more clear